### PR TITLE
View/advanced search form

### PR DESCRIPTION
**Pull request by aburgosu**

### CODE REVIEW

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
+package com.fundation.logic.view;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class AdvancedSearchPanel extends JPanel {
+    private JTextField textFieldVideoBitrate;
+    private JTextField textFieldVideoFramerate;
+    private JTextField textFieldAudioBitrate;
+
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:32]: Comment this method.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    public AdvancedSearchPanel() {
+        GridBagLayout gridBagLayout = new GridBagLayout();
+        gridBagLayout.columnWidths = new int[]{66, 0, 0};
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:35]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        gridBagLayout.rowHeights = new int[]{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:36]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        gridBagLayout.columnWeights = new double[]{0.0, 1.0, Double.MIN_VALUE};
+        gridBagLayout.rowWeights = new double[]{0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, Double.MIN_VALUE};
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:38]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        setLayout(gridBagLayout);
+
+        JLabel lblVideo = new JLabel("Video");
+        lblVideo.setFont(new Font("Tahoma", Font.BOLD, 14));
+        GridBagConstraints gbc_lblVideo = new GridBagConstraints();
+        gbc_lblVideo.gridwidth = 2;
+        gbc_lblVideo.insets = new Insets(0, 0, 5, 0);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:45]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_lblVideo.gridx = 0;
+        gbc_lblVideo.gridy = 0;
+        add(lblVideo, gbc_lblVideo);
+
+        JLabel lblCodecVideo = new JLabel("Codec: ");
+        GridBagConstraints gbc_lblCodecVideo = new GridBagConstraints();
+        gbc_lblCodecVideo.anchor = GridBagConstraints.EAST;
+        gbc_lblCodecVideo.insets = new Insets(0, 0, 5, 5);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:53]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+        gridBagLayout.rowWeights = new double[]{0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, Double.MIN_VALUE};
+        setLayout(gridBagLayout);
+
+        JLabel lblVideo = new JLabel("Video");
+        lblVideo.setFont(new Font("Tahoma", Font.BOLD, 14));
+        GridBagConstraints gbc_lblVideo = new GridBagConstraints();
+        gbc_lblVideo.gridwidth = 2;
+        gbc_lblVideo.insets = new Insets(0, 0, 5, 0);
+        gbc_lblVideo.gridx = 0;
+        gbc_lblVideo.gridy = 0;
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:47]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 6 above)
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:53]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        GridBagConstraints gbc_lblCodecVideo = new GridBagConstraints();
+        gbc_lblCodecVideo.anchor = GridBagConstraints.EAST;
+        gbc_lblCodecVideo.insets = new Insets(0, 0, 5, 5);
+        gbc_lblCodecVideo.gridx = 0;
+        gbc_lblCodecVideo.gridy = 1;
+        add(lblCodecVideo, gbc_lblCodecVideo);
+
+        JComboBox comboBoxVideoCodec = new JComboBox();
+        GridBagConstraints gbc_comboBoxVideoCodec = new GridBagConstraints();
+        gbc_comboBoxVideoCodec.insets = new Insets(0, 0, 5, 0);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:60]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_comboBoxVideoCodec.fill = GridBagConstraints.HORIZONTAL;
+        gbc_comboBoxVideoCodec.gridx = 1;
+        gbc_comboBoxVideoCodec.gridy = 1;
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:63]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+        GridBagConstraints gbc_lblBitrateVideo = new GridBagConstraints();
+        gbc_lblBitrateVideo.anchor = GridBagConstraints.EAST;
+        gbc_lblBitrateVideo.insets = new Insets(0, 0, 5, 5);
+        gbc_lblBitrateVideo.gridx = 0;
+        gbc_lblBitrateVideo.gridy = 2;
+        add(lblBitrateVideo, gbc_lblBitrateVideo);
+
+        textFieldVideoBitrate = new JTextField();
+        GridBagConstraints gbc_textFieldVideoBitrate = new GridBagConstraints();
+        gbc_textFieldVideoBitrate.insets = new Insets(0, 0, 5, 0);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:76]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_textFieldVideoBitrate.fill = GridBagConstraints.HORIZONTAL;
+        gbc_textFieldVideoBitrate.gridx = 1;
+        gbc_textFieldVideoBitrate.gridy = 2;
+        add(textFieldVideoBitrate, gbc_textFieldVideoBitrate);
+        textFieldVideoBitrate.setColumns(10);
+
+        JLabel lblFramerate = new JLabel("Framerate: ");
+        GridBagConstraints gbc_lblFramerate = new GridBagConstraints();
+        gbc_lblFramerate.anchor = GridBagConstraints.EAST;
+        gbc_lblFramerate.insets = new Insets(0, 0, 5, 5);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:86]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_lblFramerate.gridx = 0;
+        gbc_lblFramerate.gridy = 3;
+        add(lblFramerate, gbc_lblFramerate);
+
+        textFieldVideoFramerate = new JTextField();
+        GridBagConstraints gbc_textFieldVideoFramerate = new GridBagConstraints();
+        gbc_textFieldVideoFramerate.insets = new Insets(0, 0, 5, 0);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:93]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_textFieldVideoFramerate.fill = GridBagConstraints.HORIZONTAL;
+        gbc_textFieldVideoFramerate.gridx = 1;
+        gbc_textFieldVideoFramerate.gridy = 3;
+        add(textFieldVideoFramerate, gbc_textFieldVideoFramerate);
+        textFieldVideoFramerate.setColumns(10);
+
+        JLabel label = new JLabel("");
+        GridBagConstraints gbc_label = new GridBagConstraints();
+        gbc_label.insets = new Insets(0, 0, 5, 5);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:102]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_label.gridx = 0;
+        gbc_label.gridy = 4;
+        add(label, gbc_label);
+
+        JSeparator separator = new JSeparator();
+        GridBagConstraints gbc_separator = new GridBagConstraints();
+        gbc_separator.insets = new Insets(0, 0, 5, 5);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:109]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_separator.gridx = 0;
+        gbc_separator.gridy = 5;
+        add(separator, gbc_separator);
+
+        JLabel lblAudio = new JLabel("Audio");
+        lblAudio.setFont(new Font("Tahoma", Font.BOLD, 14));
+        GridBagConstraints gbc_lblAudio = new GridBagConstraints();
+        gbc_lblAudio.insets = new Insets(0, 0, 5, 0);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:117]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_lblAudio.gridwidth = 2;
+        gbc_lblAudio.gridx = 0;
+        gbc_lblAudio.gridy = 6;
+        add(lblAudio, gbc_lblAudio);
+
+        JLabel lblCodecAudio = new JLabel("Codec: ");
+        GridBagConstraints gbc_lblCodecAudio = new GridBagConstraints();
+        gbc_lblCodecAudio.anchor = GridBagConstraints.EAST;
+        gbc_lblCodecAudio.insets = new Insets(0, 0, 5, 5);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:126]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_lblCodecAudio.gridx = 0;
+        gbc_lblCodecAudio.gridy = 7;
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:128]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 18]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_lblAudio.gridwidth = 2;
+        gbc_lblAudio.gridx = 0;
+        gbc_lblAudio.gridy = 6;
+        add(lblAudio, gbc_lblAudio);
+
+        JLabel lblCodecAudio = new JLabel("Codec: ");
+        GridBagConstraints gbc_lblCodecAudio = new GridBagConstraints();
+        gbc_lblCodecAudio.anchor = GridBagConstraints.EAST;
+        gbc_lblCodecAudio.insets = new Insets(0, 0, 5, 5);
+        gbc_lblCodecAudio.gridx = 0;
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:127]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 19]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_lblCodecAudio.gridy = 7;
+        add(lblCodecAudio, gbc_lblCodecAudio);
+
+        JComboBox comboBoxAudioCodec = new JComboBox();
+        GridBagConstraints gbc_comboBoxAudioCodec = new GridBagConstraints();
+        gbc_comboBoxAudioCodec.insets = new Insets(0, 0, 5, 0);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:133]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 20]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_comboBoxAudioCodec.fill = GridBagConstraints.HORIZONTAL;
+        gbc_comboBoxAudioCodec.gridx = 1;
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:135]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 21]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_comboBoxAudioCodec.gridy = 7;
+        add(comboBoxAudioCodec, gbc_comboBoxAudioCodec);
+
+        JLabel lblChannels = new JLabel("Channels: ");
+        GridBagConstraints gbc_lblChannels = new GridBagConstraints();
+        gbc_lblChannels.anchor = GridBagConstraints.EAST;
+        gbc_lblChannels.insets = new Insets(0, 0, 5, 5);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:142]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 22]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_comboBoxAudioChannels.insets = new Insets(0, 0, 5, 0);
+        gbc_comboBoxAudioChannels.fill = GridBagConstraints.HORIZONTAL;
+        gbc_comboBoxAudioChannels.gridx = 1;
+        gbc_comboBoxAudioChannels.gridy = 8;
+        add(comboBoxAudioChannels, gbc_comboBoxAudioChannels);
+
+        JLabel lblBitrate = new JLabel("Bitrate: ");
+        GridBagConstraints gbc_lblBitrate = new GridBagConstraints();
+        gbc_lblBitrate.anchor = GridBagConstraints.EAST;
+        gbc_lblBitrate.insets = new Insets(0, 0, 5, 5);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:158]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java [hunk 23]
```diff
@@ -0,0 +1,158 @@
...
+        gbc_lblBitrate.gridx = 0;
+        gbc_lblBitrate.gridy = 9;
+        add(lblBitrate, gbc_lblBitrate);
+
+        textFieldAudioBitrate = new JTextField();
+        GridBagConstraints gbc_textFieldAudioBitrate = new GridBagConstraints();
+        gbc_textFieldAudioBitrate.insets = new Insets(0, 0, 5, 0);
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:165]: Magical numbers.

File: src/main/java/com/fundation/logic/view/LoadSavePanel.java [hunk 24]
```diff
@@ -0,0 +1,63 @@
...
+    private JTable dataTable;
+
+    /**
+     * Create the panel.
+     */
+    public LoadSavePanel() {
+        GridBagLayout gridBagLayout = new GridBagLayout();
+        gridBagLayout.columnWidths = new int[]{121, 70, 0, 70, 0, 70, 91, 0};
+        gridBagLayout.rowHeights = new int[]{0, 0, 0};
+        gridBagLayout.columnWeights = new double[]{1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, Double.MIN_VALUE};
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/LoadSavePanel.java:27]: Magical numbers.

File: src/main/java/com/fundation/logic/view/LoadSavePanel.java [hunk 25]
```diff
@@ -0,0 +1,63 @@
...
+
+public class LoadSavePanel extends JPanel {
+    private JTable dataTable;
+
+    /**
+     * Create the panel.
+     */
+    public LoadSavePanel() {
+        GridBagLayout gridBagLayout = new GridBagLayout();
+        gridBagLayout.columnWidths = new int[]{121, 70, 0, 70, 0, 70, 91, 0};
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/LoadSavePanel.java:25]: Magical numbers.

File: src/main/java/com/fundation/logic/view/LoadSavePanel.java [hunk 26]
```diff
@@ -57,7 +63,5 @@ public LoadSavePanel() {
         gbc_btnDelete.gridx = 5;
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/LoadSavePanel.java:63]: Magical numbers.

File: src/main/java/com/fundation/logic/view/LoadSavePanel.java [hunk 27]
```diff
@@ -57,7 +63,5 @@ public LoadSavePanel() {
...
         gbc_btnDelete.gridy = 1;
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/logic/view/LoadSavePanel.java:64]: Magical numbers.

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 2 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:35]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 4 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:38]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 5 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:45]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 6 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:53]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:53]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 8 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:60]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 9 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:63]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:63]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 11 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:86]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 16 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:126]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 20 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:135]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 21 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:142]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 22 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:158]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java (hunk 23 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/AdvancedSearchPanel.java:165]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

File: src/main/java/com/fundation/logic/view/LoadSavePanel.java (hunk 25 above)
Comment (aburgosu) [reply src/main/java/com/fundation/logic/view/LoadSavePanel.java:25]: Numbers of coordinates in GridBagLayout, please refactor in the next Sprint

**Pull request merged**